sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -323,11 +374,18 @@ private[joins] object UnsafeHashedRelation {
     // Create a mapping of buildKeys -> rows
     val keyGenerator = UnsafeProjection.create(key)
     var numFields = 0
+    val numKeys = key.length
+    val isOriginalInputEmpty = !input.hasNext
+    var allNullColumnKeyExistsInOriginalInput: Boolean = false
     while (input.hasNext) {
       val row = input.next().asInstanceOf[UnsafeRow]
       numFields = row.numFields()
       val key = keyGenerator(row)
-      if (!key.anyNull) {
+      if ((0 until numKeys).forall(key.isNullAt)) {
```

Comment:
I think you should only do this check if isNullAware is true. Why pay the perf penalty of checking the row again otherwise ?

Reply:
sure

Reply:
how about
```
if (isNullAware &&
        !allNullColumnKeyExistsInOriginalInput &&
        (0 until numKeys).forall(key.isNullAt)) {
        allNullColumnKeyExistsInOriginalInput = true
      }
```

Reply:
works for me :-)

Reply:
done